### ISSUE TITLE
ManyToOneRel reverse fields are causing extra queries

### ISSUE THREAD

**Issue by yardensachs** [tfoxy/graphene-django-optimizer#13]
When querying for a reverse foreign key field (ManyToOneRel), the prefetch does not add the original pointer field. This causes Django to query that field for each iteration.

Here is an example of the issue:
```
{
  allClients {
    products {
      id
    }
  }
}
```

Issue/Actual issue:
```sql
SELECT "client"."id",
       "client"."deleted",
FROM "client"

SELECT "product"."id"
FROM "product"
WHERE "product"."client_id" IN ('d4b3aab3-aac2-44c3-aed3-504d315d52c7'::uuid,
                                '1fc41289-cefd-4fc0-9fa1-e6247d84f4ee'::uuid)

SELECT "product"."id",
       "product"."client_id"
FROM "product"
WHERE "product"."id" = '92391767-9c60-4d0b-9b01-1b050b0e9573'::uuid

SELECT "product"."id",
       "product"."client_id"
FROM "product"
WHERE "product"."id" = '1002b6ab-741a-4f5a-8bd4-f9d1b218d0c0'::uuid

SELECT "product"."id",
       "product"."client_id"
FROM "product"
WHERE "product"."id" = '1810afee-e931-45b8-a1e0-9be95a39abb8'::uuid

```

Expected queries:
```sql

SELECT "client"."id",
       "client"."deleted",
FROM "client"

SELECT "product"."id"
       "product"."client_id"
FROM "product"
WHERE "product"."client_id" IN ('d4b3aab3-aac2-44c3-aed3-504d315d52c7'::uuid,
                                '1fc41289-cefd-4fc0-9fa1-e6247d84f4ee'::uuid)
```

Here is some code to show the issue with query counting:

```python
# Models

class RelatedOneToManyItem(models.Model):
    name = models.CharField(max_length=100, blank=True)
    item = models.ForeignKey(Item, on_delete=models.PROTECT, related_name='otm_items')

# Schema

class RelatedOneToManyItemType(DjangoObjectType):
    class Meta:
        model = RelatedOneToManyItem

# Test

from django.test.utils import CaptureQueriesContext
from django.db import connection

@pytest.mark.django_db
def test_should_check_reverse_relations_add_foreign_key():
    info = create_resolve_info(schema, '''
        query {
            items {
                otmItems {
                    id
                }
            }
        }
    ''')
    qs = Item.objects.all()
    items = gql_optimizer.query(qs, info)
    optimized_items = qs.prefetch_related(
        Prefetch(
            'otm_items',
            queryset=RelatedOneToManyItem.objects.only('id', 'item_id'),
        ),
    )
    assert_query_equality(items, optimized_items)

    for i in range(10):
        the_item = Item.objects.create(name='foo')
        for k in range(10):
            related_item = RelatedOneToManyItem.objects.create(name='bar{}{}'.format(i,k), item=the_item)

    with CaptureQueriesContext(connection) as expected_query_capture:
        for i in items:
            for k in i.otm_items.all():
                pass

    with CaptureQueriesContext(connection) as optimized_query_capture:
        for i in optimized_items:
            for k in i.otm_items.all():
                pass

    assert len(optimized_query_capture.captured_queries) == 2
    assert len(expected_query_capture) == len(optimized_query_capture)

```

I was able to resolve this by adding this:

```diff
        if model_field.one_to_many or model_field.many_to_many:
            field_store = self._optimize_gql_selections(
                self._get_type(field_def),
                selection,
                # parent_type,
            )

+            if isinstance(model_field, ManyToOneRel):
+                field_store.only(model_field.field.name)

            related_queryset = model_field.related_model.objects.all()
            store.prefetch_related(name, field_store, related_queryset)
            return True
```

Under here:
https://github.com/tfoxy/graphene-django-optimizer/blob/a2522943016f58dd250f3f9ea6a88af539a94bbb/graphene_django_optimizer/query.py#L179

I was unable to have the other tests pass though. not sure what would be the issue.

**Comment by yardensachs** [comment]
See my PR #14

Need some help there

**Comment by abtuleshov** [comment]
in the example in docs we have:
`categories = (
    Category.objects
    .only('id', 'name')
    .prefetch_related(Prefetch(
        'ingredients',
        queryset=Ingredient.objects.only('id', 'name'),
    ))
)`

The decision of this issue is add the 'category' field to Prefetch:
`categories = (
    Category.objects
    .only('id', 'name')
    .prefetch_related(Prefetch(
        'ingredients',
        queryset=Ingredient.objects.only('id', 'name', 'category'),
    ))
)`

**Comment by tfoxy** [comment]
This is being solved in #16 . Thanks @yardensachs for the detailed explanation and PR.

**Comment by tfoxy** [comment]
Fixed in #16!
